plugins/inputs/t128_lte/t128_lte.go

```diff
@@ -0,0 +1,340 @@
+package t128_lte
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const (
+	T128_CONFIG_FILE = "/var/run/128technology/config.json"
+	LTE_PATH_DIR     = "/var/run/128technology/lte"
+	GLOBAL_INIT_PATH = "/etc/128technology/global.init"
+	LOCAL_INIT_PATH  = "/etc/128technology/local.init"
+)
+
+var SERVICE_STATUS_MAP = map[string]string{
+	"not-registered-searching": "Unable to Register",
+	"no-service":               "No Service Available",
+	"authentication-failed":    "Authentication Failure",
+	"plmn-not-allowed":         "Forbidden PLMN",
+}
+
+type Collection []interface{}
+
+type DefaultConfig struct {
+	Init struct {
+		RouterName string `json:"routerName"`
+		Id         string `json:"id"`
+	} `json:"init"`
+}
+
+type Nodes struct {
+	Router []struct {
+		Name string `json:"name"`
+		Node []struct {
+			Name string `json:"name"`
+		} `json:"node"`
+	} `json:"router"`
+}
+
+type DeviceInterfaceDetails struct {
+	Config struct {
+		Authority struct {
+			Router []struct {
+				Node []struct {
+					DeviceInterface []struct {
```

Comment:
It seems like you'll need to split these out into separate structs because you need to reference the structs to pass config around throughout the plugin.